services/ux-backend/handlers/expandstorage/handler.go

```diff
@@ -0,0 +1,310 @@
+package expandstorage
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strconv"
+
+	ocsv1 "github.com/red-hat-storage/ocs-operator/api/v4/v1"
+	storagecluster "github.com/red-hat-storage/ocs-operator/v4/controllers/storagecluster"
+	"github.com/red-hat-storage/ocs-operator/v4/services/ux-backend/handlers"
+	cephv1 "github.com/rook/rook/pkg/apis/ceph.rook.io/v1"
+	corev1 "k8s.io/api/core/v1"
+	storagev1 "k8s.io/api/storage/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/klog/v2"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func HandleMessage(w http.ResponseWriter, r *http.Request, client client.Client, namespace string) {
+	switch r.Method {
+	case "POST":
+		handlePost(w, r, client, namespace)
+	default:
+		handleUnsupportedMethod(w, r)
+	}
+}
+
+func handlePost(w http.ResponseWriter, r *http.Request, client client.Client, namespace string) {
+	// When ContentLength is 0 that means request body is empty and
+	var err error
+	if r.ContentLength == 0 {
+		klog.Errorf("body in the request is required")
+		http.Error(w, "body in the request is required", http.StatusBadRequest)
+		return
+	}
+
+	type poolDetails struct {
+		VolumeType           string `json:"volumeType"`
+		PoolName             string `json:"poolName"`
+		DataProtectionPolicy int    `json:"dataProtectionPolicy"`
+		EnableCompression    bool   `json:"enableCompression"`
+		FilesystemName       string `json:"filesystemName"`
+		FailureDomain        string `json:"failureDomain"`
+	}
+
+	type storageClassDetails struct {
+		ReclaimPolicy                string `json:"reclaimPolicy"`
+		Name                         string `json:"name"`
+		VolumeBindingMode            string `json:"volumeBindingMode"`
+		EnableStorageClassEncryption bool   `json:"enableStorageClassEncryption"`
+		EncryptionKMSID              string `json:"encryptionKMSID"`
+	}
+
+	var ExpandStorage = struct {
+		StorageClassForOSDs string              `json:"storageClassForOSDs"`
+		EnableEncryption    bool                `json:"enableEncryption"`
+		Storage             string              `json:"storage"`
+		Replica             int                 `json:"replica"`
+		Count               int                 `json:"count"`
+		StorageClusterName  string              `json:"storageClusterName"`
+		PoolDetails         poolDetails         `json:"poolDetails"`
+		StorageClassDetails storageClassDetails `json:"storageClassDetails"`
+	}{}
+
+	if err = json.NewDecoder(r.Body).Decode(&ExpandStorage); err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	storageCluster := &ocsv1.StorageCluster{}
+	err = client.Get(r.Context(), types.NamespacedName{Name: ExpandStorage.StorageClusterName, Namespace: namespace}, storageCluster)
```

Comment:
Is this only supported for storagecluster in operator namespace?

Reply:
Yes we do run storage cluster on same namespace as per this feature